app/org/maproulette/models/dal/TaskDAL.scala

```diff
@@ -194,13 +194,22 @@ class TaskDAL @Inject()(override val db: Database,
       val geometries = (value \ "geometries").asOpt[String].getOrElse(cachedItem.geometries)
       val changesetId = (value \ "changesetId").asOpt[Long].getOrElse(cachedItem.changesetId.getOrElse(-1L))
 
-      this.mergeUpdate(cachedItem.copy(name = name,
+      val task = this.mergeUpdate(cachedItem.copy(name = name,
         parent = parentId,
         instruction = Some(instruction),
         status = Some(status),
         geometries = geometries,
         priority = priority,
         changesetId = Some(changesetId)), user)
+
+      if (status == Task.STATUS_CREATED) {
```

Comment:
So it might be a little more efficient to have a scheduled job, update the status for the challenge like once a day or something like that, instead of having this call made to the database on every single task update. Unless you need to have the "finished" status immediately.

Reply:
We do need the "finished" status immediately to know whether to show the user a "challenge complete" message.